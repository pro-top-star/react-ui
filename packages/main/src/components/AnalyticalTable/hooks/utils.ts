export const makeTemplateColumns = (columns, resizedColumns) => {
  return columns
    .map((column) => {
      if (resizedColumns[column.id]) {
        return `${resizedColumns[column.id]}px`;
      }
      return `minmax(${column.minWidth}px, ${column.width || '1fr'})`;
    })
    .join(' ');
};
