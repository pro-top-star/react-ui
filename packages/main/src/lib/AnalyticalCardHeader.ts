import { AnalyticalCardHeader } from '../components/AnalyticalCard/header/AnalyticalCardHeader';

export { AnalyticalCardHeader };
