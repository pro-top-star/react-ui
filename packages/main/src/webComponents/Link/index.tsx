import { Event } from '@ui5/webcomponents-react-base/lib/Event';
import { LinkDesign } from '@ui5/webcomponents-react/lib/LinkDesign';
import { withWebComponent } from '@ui5/webcomponents-react/lib/withWebComponent';
import UI5Link from '@ui5/webcomponents/dist/Link';
import React, { FC } from 'react';
import { WithWebComponentPropTypes } from '../../internal/withWebComponent';

export interface LinkPropTypes extends WithWebComponentPropTypes {
  disabled?: boolean; // @generated
  href?: string; // @generated
  target?: string; // @generated
  design?: LinkDesign; // @generated
  wrap?: boolean; // @generated
  onClick?: (event: Event) => void; // @generated
  children?: string; // @generated
}

/**
 * <code>import { Link } from '@ui5/webcomponents-react/lib/Link';</code>
 * <br />
 * <a href="https://sap.github.io/ui5-webcomponents/playground/components/Link" target="_blank">UI5 Web Components Playground</a>
 */
const Link: FC<LinkPropTypes> = withWebComponent<LinkPropTypes>(UI5Link);

Link.displayName = 'Link';

Link.defaultProps = {
  design: LinkDesign.Default // @generated
};

export { Link };
