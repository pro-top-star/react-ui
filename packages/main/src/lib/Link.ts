import { Link } from '../webComponents/Link';

export { Link };
