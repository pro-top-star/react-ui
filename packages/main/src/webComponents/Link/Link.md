[Link to UI5 Web Components Playground](https://sap.github.io/ui5-webcomponents/playground/components/Link)

```jsx
import { Link } from '@ui5/webcomponents-react/lib/Link';
```
