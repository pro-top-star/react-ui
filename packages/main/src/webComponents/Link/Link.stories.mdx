import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import { Link } from '@ui5/webcomponents-react/lib/Link';
import { LinkDesign } from '@ui5/webcomponents-react/lib/LinkDesign';
import { CSSProperties, Ref } from 'react';
import { createSelectArgTypes } from '@shared/stories/createSelectArgTypes';
import { DocsHeader } from '@shared/stories/DocsHeader';
import { DocsCommonProps } from '@shared/stories/DocsCommonProps';

<Meta
  title="Data Display / Link"
  component={Link}
  argTypes={{
    ...createSelectArgTypes({ design: LinkDesign }),
    children: { control: 'text' },
    ...DocsCommonProps
  }}
  args={{
    design: LinkDesign.Default,
    children: 'Link Text',
    style: {},
    className: '',
    tooltip: '',
    slot: '',
    ref: null
  }}
/>

<DocsHeader />

<br />

## Example

<Canvas>
  <Story name="Default">
    {(args) => {
      return <Link {...args} />;
    }}
  </Story>
</Canvas>

## Properties

<ArgsTable story="." />
<div style={{fontFamily:'var(--sapFontFamily)', fontSize:'var(--sapFontSize)', color:'var(--sapTextColor)'}}>
<h3>Usage</h3>


You can set the <code>Link</code> to be enabled or disabled. <br /><br />
To create a visual hierarchy in large lists of links, you can set the less important links as <code>Subtle</code> or the
more important ones as <code>Emphasized</code>, by using the <code>design</code> property. <br /><br />
If the <code>href</code> property is set, the link behaves as the basic HTML anchor tag and opens the specified URL in
the given target frame (<code>target</code> property). To specify where the linked content is opened, you can use the

<code>target</code> property.
</div>

