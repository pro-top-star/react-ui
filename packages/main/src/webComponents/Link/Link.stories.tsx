import { Link } from '@ui5/webcomponents-react/lib/Link';
import { LinkDesign } from '@ui5/webcomponents-react/lib/LinkDesign';
import React from 'react';
import { createSelectArgTypes } from '@shared/stories/createSelectArgTypes';

export default {
  title: 'UI5 Web Components / Link',
  component: Link,
  argTypes: {
    ...createSelectArgTypes({ design: LinkDesign }),
    children: {
      type: null
    },
    ref: {
      type: null
    }
  },
  args: {
    design: LinkDesign.Default
  }
};

export const generatedDefaultStory = (props) => (
  <Link
    design={props.design}
    disabled={props.disabled}
    href={props.href}
    target={props.target}
    wrap={props.wrap}
    onClick={props.onClick}
  >
    Some Content
  </Link>
);

generatedDefaultStory.storyName = 'Generated default story';
