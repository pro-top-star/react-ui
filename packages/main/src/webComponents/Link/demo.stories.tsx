import { action } from '@storybook/addon-actions';
import { boolean, select } from '@storybook/addon-knobs';
import { Link } from '@ui5/webcomponents-react/lib/Link';
import { LinkDesign } from '@ui5/webcomponents-react/lib/LinkDesign';
import React from 'react';
import notes from './Link.md';

export default {
  title: 'UI5 Web Components / Link',
  component: Link,
  parameters: { notes }
};

export const generatedDefaultStory = () => (
  <Link
    disabled={boolean('disabled', false)}
    href={'generatedString'}
    target={'generatedString'}
    design={select('design', LinkDesign, null)}
    wrap={boolean('wrap', false)}
    onClick={action('onClick')}
  >
    Some Content
  </Link>
);

generatedDefaultStory.story = {
  name: 'Generated default story'
};
