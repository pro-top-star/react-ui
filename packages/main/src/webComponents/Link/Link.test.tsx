import { mountThemedComponent } from '@shared/tests/utils';
import { Link } from '@ui5/webcomponents-react/lib/Link';
import React from 'react';

describe('Link', () => {
  test('Basic Test (generated)', () => {
    const wrapper = mountThemedComponent(<Link />);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
