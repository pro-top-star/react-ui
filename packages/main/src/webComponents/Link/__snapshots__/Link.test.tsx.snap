// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Basic Test (generated) 1`] = `
<ui5-link
  design="Default"
/>
`;
